src/root/components/ListView/index.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import Message, {
+  Props as MessageProps
+} from './Message';
+import List, {
+  Props as ListProps,
+  GroupCommonProps,
+  OptionKey
+} from '../List';
+
+type MessagePropOmission = 'className' | 'message' | 'empty' | 'pending' | 'errored';
+
+export type Props<
+  D,
+  P,
+  K extends OptionKey,
+  GP extends GroupCommonProps,
+  GK extends OptionKey
+  > = ListProps<D, P, K, GP, GK> &
+  Omit<MessageProps, MessagePropOmission> & {
+    className?: string;
+    errored: boolean;
+    pending: boolean;
+  };
+
+function ListView<
+  D,
+  P,
+  K extends OptionKey,
+  GP extends GroupCommonProps,
+  GK extends OptionKey
+>(props: Props<D, P, K, GP, GK>) {
+  const {
+    className,
+    errored,
+    pending,
+    data,
+    emptyMessage,
+    pendingMessage,
+    erroredMessage,
+    ...otherListProps
+  } = props;
+  const empty = !(data?.length && data.length > 0);
```

Comment:
I guess we could make this a bit simpler by writing it as :-
const empty = !(data && data.length > 0 );

Reply:
This logic is picked from deep-ui list view.